### PR TITLE
Remove basepath from cachefile name

### PR DESCRIPTION
**Pull request by Casmo**
Removing basepath from the filename when working with subdirectories.

### CODE REVIEW

**Comment by dereuromark** [comment]
CS looks off.
Do we have a test case here?

**Comment by Casmo** [comment]
I'm not sure how I would make test with different subdirectories as base. But here is the difference in `Controller->request`:

``` php
// Seen from an app places in the root (/).
    url => false
    base => ''
    webroot => '/'
    here => '/'

// Seen from an app places in a subdirectory (/cms).
    url => false
    base => '/cms'
    webroot => '/cms/'
    here => '/cms/'
```

My guess the Component test would look like this.

``` php
    public function testFileFromSubdirection()
    {
        $this->Controller->request->url = 'pages/view/1';
        $this->Controller->request->base = '/myapp';
        $this->Controller->request->webroot = '/myapp/';
        $this->Controller->request->here = '/myapp/pages/view/1';

        $this->Controller->response = $this->getMock('Cake\Network\Response', ['body', 'type']);

        $this->Controller->response->expects($this->once())
            ->method('body')
            ->will($this->returnValue('Foo bar'));
        $this->Controller->response->expects($this->once())
            ->method('type')
            ->will($this->returnValue('application/json'));

        $event = new Event('Controller.shutdown', $this->Controller);
        $this->Controller->Cache->shutdown($event);
        $file = CACHE . 'views' . DS . 'pages-view-1.html';
        $this->assertEquals(is_file($file), true);

        unlink($file);
    }
```

File: src/Controller/Component/CacheComponent.php [hunk 1]
```diff
@@ -86,6 +86,7 @@ protected function _writeFile($content, $duration) {
 		}
 
 		$path = $this->request->here();
+        $path = str_replace($this->request->base, '', $path);
```
Comment (dereuromark) [inline src/Controller/Component/CacheComponent.php:89]: cs errors

**Comment by dereuromark** [comment]
Whats the status here?

**Pull request closed without merging**